things-to-do

```diff
@@ -0,0 +1,19 @@
+array data type -> support for null (not empty!) array
+
+proper serialization of request depending on api_version (some versions serialize only some fields)
+
+request decoder: missing request types (right now we only have Produce, Fetch, etc.)
+
+requests in general - kafka 2.0 has 43 types of request (0.11 has ~30)
+	will it be okay to macro out some of the definitions?
+	compare https://github.com/apache/kafka/blob/2.0/clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java#L271
+
+request decoder: do we need to capture payload every time?
+	for broker-proxy this might not be necessary (in the simplest model of just counting requests)
+	for cluster-proxy we would need that - need configurable deserializer (or somesuch) that would create correct buffer
+		(typical max kafka request is 1mb in size)
+
+whole response codec (should be very similar to request)
+
+serialization using Encoder can overflow (whole api just takes char* as argument)
+	(the tests are full of char[1024] too)
```

Comment:
do you have any idea how would you prefer to serialize the messages?
I can take envoy Buffer as argument if you prefer as well